### PR TITLE
feat(examples): declare the app block and ship the first template dashboard [SAP-3252]

### PR DESCRIPTION
**Pull request by gwitwer**
## Primary change type

- [ ] Bug fix
- [ ] Documentation
- [x] Feature
- [ ] Tests
- [ ] Dependency update
- [ ] Maintenance or refactor

## Problem and motivation

Templates have no way to declare a dashboard, so the template page has nothing to preview and the clone path has nothing to publish. The backend half already landed (Sapiom PR 4990, [SAP-3253](https://linear.app/sapiom/issue/SAP-3253)): its zod parses an `app` block that no template can yet write, because `template.schema.json` — the CI gate in this repo — does not know the field. This PR is the reference implementation the eleven per-template dashboard tickets (SAP-3311 … SAP-3321) copy: the schema block, its CI validation, the authoring docs, the screenshot hosting path, and one working dashboard end to end.

Linear: [SAP-3252](https://linear.app/sapiom/issue/SAP-3252/pilot-declare-the-app-block-in-the-template-schema-and-ship-the-first) · Epic: [SAP-3233](https://linear.app/sapiom/issue/SAP-3233)

## Summary and scope

**Schema** — `app` in `examples/template.schema.json`, matching the landed backend zod field for field: `name`, `entry`, `start`, `port`, `preview` required; `build` accepts a string, `null`, or omission; `additionalProperties: false`; `port` 1–65535; `entry` a trailing-slash relative directory; `preview` an absolute **`https://`** URL.

> Note on `preview`: the ticket text says "http/https". The landed zod is https-only (`/^https:\/\//` after `.url()`, with the mixed-content and injection-sink rationale in its comment), and the ticket also says to match the landed backend rather than re-derive it. An `http` preview that passed CI here would be silently dropped to "no dashboard" in prod, so the schema follows the backend.

**Checks** — two things JSON Schema cannot express readably, following the `manifest-secret-key` / `manifest-resource-seed` precedent:
- `manifest-app-preview` mirrors zod's `.url()` parse (the pattern alone lets `https://exa mple.com` through).
- `manifest-app-entry` requires the declared `entry` directory to exist in the example, so a dashboard cannot be declared without its source.

**Fixtures** — `scripts/examples-manifest-check.test.mjs` asserts both directions and the same edge cases the backend spec does: full block passes; no block passes; partial block fails naming each missing field; extra field fails; `build` null / `""` / omitted pass and a non-string fails; `preview` as http / `javascript:` / `data:` / relative / host-less fails; port 0 / 70000 / float / string fails; entry without trailing slash / absolute / `../` / hidden fails; `checkAppEntry` both ways plus a repo-wide pass over every declaring manifest.

**Pilot** — `scheduled-db-insight-report` declares `app` and ships `app/`: a dependency-free Node server (`node server.mjs`, no build step) serving one page that renders the latest report, the flagged anomaly, and the follow-up read as a bar chart. With `SAPIOM_API_KEY` + `SAPIOM_DEFINITION_ID` in the environment it reads the latest completed run over the Sapiom API (verified against production, read-only: it returned run 158211 live); without them it serves `sample-report.json`, the verbatim output of that same real zero-setup run, and the page footer says which source it is showing.

**Screenshot hosting path** — `preview.png` is committed beside `template.json` (outside `entry`, so it clones but is not uploaded into the running bundle) and referenced at `https://raw.githubusercontent.com/sapiom/sapiom-js/main/examples/<id>/preview.png` — the same host and ref production already reads `registry.json` and every manifest from. No bucket, credential, or CDN; it resolves on merge. `pnpm examples:app:screenshot <id>` starts the dashboard the way the publish step will (`build`, then `start`, inside `entry`) and captures the 1200×760 card viewport at 2×, so the next eleven capture identically. `playwright` is added as a root devDependency for that script (already in the store via `@playwright/test` in `packages/harness`; browsers are installed by the author, not CI).

**Docs** — `examples/AUTHORING.md` gains an `app` row and a "Shipping a dashboard" subsection in the `template.json` section: the fields, the runtime contract (`start` runs inside `entry`; avoid port 3000), the env the dashboard reads, the hosting path, the capture command, what CI checks, and what makes a dashboard worth shipping. Plus one checklist line.

**Deliberately not touched:** `examples/registry.json` (`app` is a manifest field; the card reads the detail page). The pilot's `index.ts` is unchanged.

**Open contract for E2 (the publish step):** the dashboard reads `SAPIOM_API_KEY`, `SAPIOM_DEFINITION_ID`, and optionally `SAPIOM_API_URL`; `start` is written relative to `entry` as the bundle root (`node server.mjs`). AUTHORING states both as what the publish step is expected to provide.

## Related work

Related issue or discussion: [SAP-3252](https://linear.app/sapiom/issue/SAP-3252) · backend half: sapiom/Sapiom#4990

## Validation

```text
pnpm examples:check                          — OK: 12 templates, 12 manifests schema-valid, 57 assets terminology-checked
node --test scripts/*.test.mjs               — 202 tests, 202 pass, 0 fail (incl. 25 new app-block fixtures)
pnpm examples:app:screenshot scheduled-db-insight-report — wrote examples/scheduled-db-insight-report/preview.png (2400×1520)
SAPIOM_DEFINITION_ID=523 PORT=4174 node server.mjs; curl /api/report — source.kind "live", run 158211, anomaly demo_orders
pnpm build && pnpm typecheck && pnpm lint    — exit 0
prettier --check <changed files>             — all pass
```

Screenshot as captured (this is `preview.png`):

Insight report dashboard

### Tests and documentation

Tests: `scripts/examples-manifest-check.test.mjs` (fixtures both directions, listed above). Docs: `examples/AUTHORING.md`, `examples/scheduled-db-insight-report/README.md`, `examples/scheduled-db-insight-report/app/README.md`.

## Compatibility and release impact

- Breaking or externally visible changes: On merge, production reads the pilot's manifest at `main` and its detail page gains the dashboard card once the cache TTL expires. A malformed block is contained by the backend's block-scoped `.catch` (no dashboard, manifest intact). No published package changes.
- Changeset: N/A — `examples/` and `scripts/` are outside the published packages.

## Security

- [x] I have not included secrets, credentials, private data, or unsanitized logs.
- [x] This pull request does not publicly disclose a suspected vulnerability. I
      will follow the
      [Security Policy](https://github.com/sapiom/sapiom-js/security/policy) for
      private reporting.

## AI assistance

- [ ] I did not use AI assistance for this change.
- [x] I used AI assistance and have described it below.

Authored with Claude Code (Fable 5.1) in an AO-managed session: it read the landed backend zod and its spec, wrote the schema block, checks, fixtures, dashboard, and docs, captured the screenshot, and ran every command listed under Validation.

## Checklist

- [x] I read `CONTRIBUTING.md`, and this contribution follows the direct-PR or issue-first policy.
- [x] This pull request addresses one focused problem and contains no unrelated cleanup.
- [x] I added or updated tests, or explained above why tests are not applicable.
- [x] I ran the relevant build, typecheck, lint, and test commands, or explained
      any N/A checks above.
- [x] I updated documentation for user-facing changes, or marked it N/A above.
- [x] I added a Changeset for a published-package change, or explained why it is not applicable.
- [x] I can explain and maintain every submitted change, including any AI-assisted work.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01FoizcZ4qFbrxAGqZ6QmMsa

## Summary by CodeRabbit

- **New Features**
  - Added an optional dashboard to the scheduled database insight report example.
  - Dashboard displays reports, anomaly details, follow-up activity charts, delivery status, and source metadata.
  - Supports live report data when configured, with captured sample data as a fallback.
  - Added light and dark themes plus responsive layouts.

- **Documentation**
  - Documented dashboard configuration, required manifest fields, validation rules, and authoring steps.
  - Added usage and data-source documentation for the example dashboard.

- **Validation**
  - Added checks for valid dashboard directories, preview URLs, and manifest configuration.
  - Added automated screenshot capture at the required preview dimensions.

### CODE REVIEW

File: examples/scheduled-db-insight-report/app/server.mjs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  "",
+);
+const API_KEY = process.env.SAPIOM_API_KEY || "";
+const DEFINITION_ID = process.env.SAPIOM_DEFINITION_ID || "";
+const LIVE_CACHE_MS = 30_000;
+
+/** The captured run, read once — it never changes while the server is up. */
+const sample = readFile(path.join(HERE, "sample-report.json"), "utf8").then(
+  (text) => JSON.parse(text),
+);
```
[the remark this answers is not in the document]
Comment (gwitwer) [reply examples/scheduled-db-insight-report/app/server.mjs:72]: Fixed in 55407c53. Both module-level reads (`sample` and `page`) now attach a `.catch(() => {})` so the rejection is observed at creation; the request handler still awaits the original promise and returns a 500 with the error text on the matching route.

File: examples/scheduled-db-insight-report/app/server.mjs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+let liveCache = { at: 0, value: null };
+
+async function api(route) {
+  const res = await fetch(`${API_URL}${route}`, {
+    headers: { "x-api-key": API_KEY, accept: "application/json" },
+    signal: AbortSignal.timeout(10_000),
+  });
```
[the remark this answers is not in the document]
Comment (gwitwer) [reply examples/scheduled-db-insight-report/app/server.mjs:85]: Fixed in 55407c53. The credentialed fetch now uses `redirect: "manual"`; the API does not redirect its JSON routes, so a 3xx is treated as a failed read (`ok` is false) and falls back to the captured run. On the transport side, a non-https `SAPIOM_API_URL` is refused at startup when the key is set, with one deliberate carve-out: a loopback host (`localhost`, `127.0.0.1`, `::1`) over http stays allowed, because that is where a local Sapiom backend runs and the harness e2e flow points templates at it. The refusal is logged once and surfaces in the page footer as the reason live mode is off. Verified all four paths by hand: no env → captured; https prod → live; http remote → refused with the message; http loopback → live mode on.

**Comment by gwitwer** [comment]
Re the nitpick on `scripts/examples-app-screenshot.mjs` (process tree shutdown): addressed in 55407c53. The dashboard is spawned `detached` on POSIX so the shell and `node server.mjs` share a process group, `stop()` signals the group with `process.kill(-pid)` and awaits the child's `close` event before the script returns. On Windows there is no process group, so the shell is signalled as before and the child is left to exit on its own; `taskkill /T` is out of scope for a dev-only helper this ticket ships, and the comment in the file says so. Verified: after a recapture no listener remains on the dashboard's port.

**Review by ynadge — DISMISSED** [review]

File: examples/AUTHORING.md [hunk 3]
```diff
@@ -506,6 +507,88 @@ means a real run — nothing pretends.
...
+
+Have your page set `<body data-ready="true">` once its data has rendered — the helper waits
+for it, so the capture is of content rather than a loading state. Look at the PNG before you
+commit it: it should read as a real product screen with real numbers in it. Regenerate it
+whenever the page changes.
+
+**What CI checks.** `pnpm examples:check` fails an `app` block that is missing a required
+field, carries a field the schema does not declare, has an `entry` without a trailing slash
+or not on disk, a `port` outside 1–65535, a `build` that is not a string, or a `preview` that
+is not an absolute `https://` URL. It accepts a manifest with no `app` block. The schema is a
```
[the remark this answers is not in the document]
Comment (gwitwer) [reply examples/AUTHORING.md]: Fixed in 76900a3d, wording taken as suggested: a `build` that is neither a string nor `null` fails. The schema and the fixtures already accepted `null`; only the sentence was wrong.

File: examples/scheduled-db-insight-report/app/server.mjs [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+// Same observed-rejection pattern as `sample`: a missing page is a 500 on `/`,
+// not a crash at startup.
+const page = readFile(path.join(HERE, "index.html"));
+page.catch(() => {});
+
+const server = createServer(async (req, res) => {
+  const url = new URL(
+    req.url ?? "/",
+    `http://${req.headers.host ?? "localhost"}`,
+  );
```
[the remark this answers is not in the document]
Comment (gwitwer) [reply examples/scheduled-db-insight-report/app/server.mjs]: Fixed in 76900a3d. The URL is now parsed inside `try` against the fixed base `http://localhost` (only the pathname is read), so a malformed Host header gets the handler's 500 path instead of rejecting the async listener. Verified by sending `Host: [` and `Host: %%bad` raw over a socket: both answered 200 for a valid path and the server kept serving.

File: scripts/examples-app-screenshot.mjs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  });
+}
+
+/** Signal the whole process group and wait for the child to actually close. */
+function stop(child) {
+  if (child.exitCode !== null || child.signalCode !== null)
+    return Promise.resolve();
+  const closed = new Promise((resolve) => child.once("close", resolve));
+  try {
+    if (process.platform === "win32") child.kill("SIGTERM");
```
[the remark this answers is not in the document]
Comment (gwitwer) [reply scripts/examples-app-screenshot.mjs]: Fixed in 76900a3d. The Windows branch of `stop()` now runs `taskkill /pid <shell pid> /T /F` so the `app.start` child goes with the shell, and the helper still awaits the child's `close` before returning. POSIX keeps the detached process group and `process.kill(-pid)`. Re-ran the capture on macOS: no listener left on the dashboard port. I have no Windows machine here, so the taskkill branch is verified by reading, not by running.

**Comment by gwitwer** [comment]
Nitpick on the AUTHORING.md fenced URL block (MD040): added the `text` language in 76900a3d.

**Review by ynadge — APPROVED** [review]
